tools/varint/src/main/java/org/apache/polaris/tools/varint/VarInt.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.tools.varint;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.primitives.Ints;
+import java.nio.ByteBuffer;
+
+/** Utility class to de-serialize <em>positive</em> integer values in a space efficient way. */
+public final class VarInt {
+  private VarInt() {}
+
+  public static int varIntLen(long v) {
+    checkArgument(v >= 0);
+    int l = 0;
+    while (true) {
+      l++;
+      if (v <= 0x7f) {
+        return l;
+      }
+      v >>= 7;
+    }
+  }
+
+  public static ByteBuffer putVarInt(ByteBuffer b, long v) {
```

Comment:
I don't know how I feel about maintaining this code. Can we use a [library](https://javadoc.io/doc/com.esotericsoftware/kryo/5.1.1/com/esotericsoftware/kryo/io/Output.html#writeLong(long,boolean)) for this?